src/libmongoc/tests/test-mongoc-topology.c

```diff
@@ -1433,28 +1529,32 @@ _test_ismaster_retry_pooled (bool hangup, int n_failures)
    /* retry immediately (for testing, "immediately" means less than 250ms */
    request = mock_server_receives_ismaster (server);
    ASSERT_CMPINT64 (bson_get_monotonic_time () - t, <, (int64_t) 250 * 1000);
+   /* Since connection was established successfully, the server description is
+    * not marked as Unknown until after a failed retry attempt. */
+   BSON_ASSERT (has_known_server (client));
```

Comment:
The previous behavior was incorrect, a failure would mark as Unknown before attempting a retry: https://jira.mongodb.org/browse/CDRIVER-3529